src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_pe/StartGame.java

```diff
@@ -123,11 +124,7 @@ public StartGame(ConnectionImpl connection) {
 		//Set PE gamemode.
 		AttributesCache attrscache = cache.getAttributesCache();
 		attrscache.setPEGameMode(gamemode);
-		//fake chunks with position, because pe doesn't like spawning in no chunk world
-		ChangeDimension.addFakeChunksAndPos(version, player, attrscache.getPEFakeSetPositionY(), packets);
-		//add two dimension switches to make sure that player ends up in right dimension even if bungee dimension switch on server switch broke stuff
-		ChangeDimension.create(version, dimension != Environment.OVERWORLD ? Environment.OVERWORLD: Environment.THE_END, player, attrscache.getPEFakeSetPositionY(), packets);
-		ChangeDimension.create(version, dimension, player, attrscache.getPEFakeSetPositionY(), packets);
+		CustomPayload.create(version, "ps:clientlock", Unpooled.EMPTY_BUFFER, packets); // lock client bound packet queue until LocalPlayerInitialised or bungee confirm
```

Comment:
This seems like a good idea, but implementation wise I'd like to see it a bit different. You can see we have some abstraction in place for the custom payload if you look at the chunk packet for instance.